AstSemantics.md

```diff
@@ -232,12 +242,49 @@ operator may be added. However, due to normal fragmentation, applications are
 instead expected release unused physical pages from the working set using the
 [`discard`](FutureFeatures.md#finer-grained-control-over-memory) future feature.
 
+The above operators operate on the [default linear memory](#linear-memory).
+
+## Table
+
+A *table* is similar to a linear memory whose elements, instead of being bytes,
+are opaque values of a particular *table element type*. This allows the table to
+contain values—like GC references, raw OS handles, or native pointers—that are
+accessed by WebAssembly code indirectly through an integer index. This feature
+bridges the gap between low-level, untrusted linear memory and high-level
+opaque handles/references at the cost of a bounds-checked table indirection.
+
+The table's element type dynamically constrains the type of elements stored 
+in the table and allows engines to avoid some type checks on table use. When
+tables are mutated, any stored value must match the element type by either a
+static validation constraint, trapping dynamic type validation check, or dynamic
+coercion (depending on where the mutation occurs).
+
+Every WebAssembly [instance](Modules.md) has one specially-designated *default*
+table which is indexed by [`call_indirect`](#calls) and other future
+table operators. Tables can either be [imported](Modules.md#imports) or 
+[defined inside the module](Modules.md#table-section), with defaultness
+indicated by a flag on the import or definition. After import or definition,
+there is no difference when calling into a table whether it was imported or
+defined internally.
+
```

Comment:
Tables of functions seems a use case worthy of special attention, and would warrant allowing multiple function tables in the MVP or well before tables of other objects.

Could this note that the table element type would be defined in the module instance. If this were not static then it would not be of user for optimize of AOT compiled code.